docs/RELEASE.md

```diff
@@ -0,0 +1,136 @@
+# Release
+
+The outputs of a release are the `polkadot` and `polkadot-parachain` nodes, runtimes for the Westend & Rococo networks, including their system parachains, and new crate versions published to `crates.io`.
+
+# Setup
+
+We have two branches: `master` and `release`. `master` is the main development branch where normal merge requests are opened. Developers need to mostly only care about this branch.  
+The `release` branch contains a version of the code that is ready to be released. Its contents are always audited. Merging to it is restricted to [Backports](#backports).
+
+# Versioning
+
+We are releasing multiple different things from this repository in one release, but 
+we don't want to use the same version for everything. Thus, in the following we explain
+the versioning story for the crates, node and Westend & Rococo. To easily refer to a release, it shall be named by its date in the form `stableYYMMDD`.
+
+## Crate
+
+We try to follow SemVer<sup>1</sup> as best as possible for versioning our crates. SemVer requires a piece of software to first declare a public API. The public API of the Polkadot SDK is hereby declared as the sum of all crates' public APIs, with the exception of the following internal crates:
+
+- @bkchr the list of node internals please
+
+The public API of our library crates is declared as all public items that are not inside a `__private` module.
+
+## Node
+
+The versioning of the node is done most of the time by only incrementing the `minor` version. 
+The `major` version is only bumped for special releases and the `patch` can be used for an 
+out of band release that fixes some critical bug. The node version is not following SemVer. 
+This means that the version doesn't express if there are any breaking changes in the CLI 
+interface or similar. The node version is declared in the `NODE_VERSION` variable in 
+`polkadot/node/primitives/src/lib.rs`.
+
+## Westend & Rococo
+
+For the these networks, in addition to incrementing the Cargo.toml version we also increment the `spec_version` and sometimes the `transaction_version`. The spec version is also following
+the node version. Its schema is: `M_mmm_ppp` and for example `1_002_000` is the node release `1.2.0`. This versioning has no further meaning, and is only done to map from an on chain `spec_version` easily to the release in this repository.  
```

Comment:
I seem to remember our versioning has historically been MMmmppxx

Reply:
There was some discussion to rather go with triplets so that it formats nicer as Rust integer, because of the underscores.  
Or do you prefer the other schema?

Reply:
Don't have a huge opinion, but we have in the past put release candidates on-chain, particularly in testnets, and this needed a version number in between two patch releases.

Reply:
Hm, we could change the spec-version to a `(u8, u8, u8, u8)` tuple with these `MMM, mmm, ppp, xxx` semantic.  
Currently we just assume one ever-increasing integer, but since its structured information, IMHO it would make sense to represent it as such.

Otherwise yea, MMmmppxx would also work.